リバーシプログラムの修正

misonabema opened the pull request:
列と行が逆になる問題を修正
まだ置いてないマスを相手の色と認識する問題を修正
お互いにおける場所がないのにゲームが終わらない問題を修正

misonabema committed:
テストプログラムが全て通るよう修正

列と行が逆になる問題を修正
まだ置いてないマスを相手の色と認識する問題を修正
お互いにおける場所がないのにゲームが終わらない問題を修正

JunichiIto commented on lib/reversi_methods.rb:
```diff
@@ -62,7 +62,7 @@ def put_stone(board, cell_ref, stone_color, dry_run: false)
   def turn(board, target_pos, attack_stone_color, direction)
     return false if target_pos.out_of_board?
-    return false if target_pos.stone_color(board) == attack_stone_color
+    return false if target_pos.stone_color(board) == attack_stone_color || target_pos.stone_color(board) == '-'
```
`'-'` は何を意味するのかぱっと見でわからないので、もう少し可読性を向上させる方法を考えてみてください。

それと、

```ruby
return false if A
return false if B || C
```

というように、条件Aだけ仲間はずれになっているのには何か意味があるのかなー？と思いました。

misonabema replied:
>条件Aだけ仲間外れになっている
不具合を見つけることだけに気が向いて失念していました！修正しました。

JunichiIto reviewed (commented):
コメントしました！

misonabema committed:
レビューの反映

'-'が既に定数`BLANK_CELL`としてあったため置き換え

```
return false if A
return false if B || C
```
となっていた部分を
```
return false if A || B || C
```
に変更

JunichiIto commented on lib/reversi_methods.rb:
```diff
@@ -61,8 +61,7 @@ def put_stone(board, cell_ref, stone_color, dry_run: false)
   def turn(board, target_pos, attack_stone_color, direction)
-    return false if target_pos.out_of_board?
-    return false if target_pos.stone_color(board) == attack_stone_color || target_pos.stone_color(board) == '-'
+    return false if target_pos.out_of_board? || target_pos.stone_color(board) == attack_stone_color || target_pos.stone_color(board) == BLANK_CELL
```
好みの問題なので、絶対にこっち、という答えはありませんが、`A || B || C` と全部横並びにするとちょっと読みづらいので、

```ruby
return false if A
return false if B
return false if C
```

と、縦並びにするのも一手かなーと思います。

また、今回の場合は単純にそうした方がdiffの行数が少なく済む、というメリットもあります。
（一般論として、diffが多ければ多いほど、予期せぬ不具合を埋め込んでしまうリスクが高まる）

JunichiIto reviewed (approved):
1点コメントしましたが大きな問題ではないのでこれでOKです！🙆‍♂️
